internal/tfr/getter.go

```diff
@@ -283,3 +281,25 @@ func httpGETAndGetResponse(ctx context.Context, getURL url.URL) ([]byte, *http.H
 	bodyData, err := ioutil.ReadAll(resp.Body)
 	return bodyData, &resp.Header, errors.WithStackTrace(err)
 }
+
+// buildRequestUrl - create url to download module using moduleRegistryBasePath
+func buildRequestUrl(registryDomain string, moduleRegistryBasePath string, modulePath string, version string) (*url.URL, error) {
+	moduleRegistryBasePath = strings.TrimSuffix(moduleRegistryBasePath, "/")
+	modulePath = strings.TrimSuffix(modulePath, "/")
+	modulePath = strings.TrimPrefix(modulePath, "/")
+
+	moduleFullPath := fmt.Sprintf("%s/%s/%s/download", moduleRegistryBasePath, modulePath, version)
+
+	if strings.HasPrefix(moduleFullPath, "https") {
```

Comment:
Since `url.Parse` will successfully parse a path, what do you think of the following alternative?:

```
moduleURL, err := url.Parse(moduleFullPath)
if err != nil {
    return nil, err
}
if moduleURL.Scheme != "" {
    return moduleURL, nil
}
return &url.URL{Scheme: "https", Host: registryDomain, Path: moduleFullPath}, nil
```

I'm not sure if the module registry would return anything that is not `https`, so might be moot, but it would at least handle cases where the registry basepath has a URL that starts with https (e.g., `httpsmodules` or something like that).